stock_picking_batch_extended/tests/test_batch.py

```diff
@@ -267,68 +274,63 @@ def test_backorder(self):
         self.assertEqual(self.product10, backorder2.move_ids.product_id)
         self.assertEqual(1, backorder2.move_ids.product_uom_qty)
         self.assertEqual(1, len(backorder2.move_line_ids))
-        self.assertEqual(1, backorder2.move_line_ids.reserved_uom_qty)
-        self.assertEqual(0, backorder2.move_line_ids.qty_done)
+        self.assertEqual(1, backorder2.move_line_ids.quantity)
 
     def test_assign_draft_pick(self):
-        picking3 = self.create_simple_picking(
+        self.picking3 = self.create_simple_picking(
             self.product11.ids, batch_id=self.batch.id
         )
-        self.assertEqual("draft", picking3.state)
+        self.assertEqual("draft", self.picking3.state)
         self.batch.action_assign()
-        context = self.batch.action_done().get("context")
-        Form(
-            self.env["stock.immediate.transfer"].with_context(**context)
-        ).save().process()
+        self.assertEqual(self.picking.state, "assigned", "Picking 1 should be ready")
+        self.assertEqual(self.picking2.state, "assigned", "Picking 2 should be ready")
+        self.assertEqual(self.picking3.state, "assigned", "Picking 3 should be ready")
+        self.picking.move_ids.write({"quantity": 5, "picked": True})
+        self.picking2.move_ids.write({"quantity": 10, "picked": True})
+        self.picking3.move_ids.write({"quantity": 5, "picked": True})
+        # There should be a wizard asking to process picking without quantity done
+        back_order_wizard_dict = self.batch.action_done()
+        self.assertTrue(back_order_wizard_dict)
+        back_order_wizard = Form.from_action(self.env, back_order_wizard_dict).save()
+        self.assertEqual(len(back_order_wizard.pick_ids), 2)
+        back_order_wizard.process()
         self.assertEqual("done", self.batch.state)
         self.assertEqual("done", self.picking.state)
         self.assertEqual("done", self.picking2.state)
-        self.assertEqual("done", picking3.state)
+        self.assertEqual("done", self.picking3.state)
 
     def test_package(self):
-        warehouse = self.browse_ref("stock.warehouse0")
```

Comment:
This test was failing. I've checked the OCA code doesn't modify the behavior of packaging, so I've rewritten the code with the standard way.